Panic happens when only Alicloud Machine Class is created without a related secret

If an Alicloud Machine Class is created and no secret for Alicloud is created, machine controller manager will panic:

panic: not support yet [recovered]
	panic: not support yet

goroutine 387 [running]:
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x17eca20, 0x1deb8c0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/gardener/machine-controller-manager/vendor/github.com/aliyun/alibaba-cloud-sdk-go/sdk.(*Client).Init(0xc42032c8c0, 0xc42032c8c0, 0xc420c78ce0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/github.com/aliyun/alibaba-cloud-sdk-go/sdk/client.go:48 +0x39
github.com/gardener/machine-controller-manager/vendor/github.com/aliyun/alibaba-cloud-sdk-go/services/ecs.NewClient(0xc420c78ce0, 0x1f, 0xc420c78ce0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/github.com/aliyun/alibaba-cloud-sdk-go/services/ecs/client.go:31 +0x40
github.com/gardener/machine-controller-manager/pkg/driver.(*AlicloudDriver).getEcsClient(0xc420c4d480, 0x1d, 0x1c6eaf7, 0x13)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_alicloud.go:315 +0x16a
github.com/gardener/machine-controller-manager/pkg/driver.(*AlicloudDriver).getVMDetails(0xc420c4d480, 0x0, 0x0, 0xc420bb7c00, 0xc4201c8ea0, 0x0, 0x0, 0xc420567330)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_alicloud.go:256 +0x1f0
github.com/gardener/machine-controller-manager/pkg/driver.(*AlicloudDriver).GetVMs(0xc420c4d480, 0x0, 0x0, 0xc42029c500, 0x14, 0x1bfc320)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_alicloud.go:283 +0x6b
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).checkMachineClass(0xc4200dcc80, 0x1bfc320, 0xc420b48600, 0xc420376280, 0xc420265a30, 0xd, 0xc42029c500, 0x14)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/machine_safety.go:351 +0x13a
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).checkAlicloudMachineClass(0xc4200dcc80)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/machine_safety.go:320 +0x125
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).checkVMObjects(0xc4200dcc80)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/machine_safety.go:158 +0x63
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).reconcileClusterMachineSafetyOrphanVMs(0xc4200dcc80, 0x0, 0x0, 0x0, 0x0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/machine_safety.go:52 +0xea
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).(github.com/gardener/machine-controller-manager/pkg/controller.reconcileClusterMachineSafetyOrphanVMs)-fm(0x0, 0x0, 0xc42000a200, 0x17eca20)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:458 +0x3e
github.com/gardener/machine-controller-manager/pkg/controller.worker.func1.1(0x1e25ce0, 0xc42000a200, 0xc4202f6f00, 0x455301, 0xf, 0x1c7bcb2, 0x1d, 0xc4203ebb00)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:496 +0xe5
github.com/gardener/machine-controller-manager/pkg/controller.worker.func1()
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:513 +0x8d
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420c4d400)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420c4d400, 0x3b9aca00, 0x0, 0x1ad3d01, 0x0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420c4d400, 0x3b9aca00, 0x0)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
github.com/gardener/machine-controller-manager/pkg/controller.createWorker.func1(0x1e25ce0, 0xc42000a200, 0x1c7bcb2, 0x1d, 0xf, 0x1, 0xc4202f6f00, 0x0, 0xc420264090)
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:475 +0x91
created by github.com/gardener/machine-controller-manager/pkg/controller.createWorker
	/Users/i041966/projects/go-root/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:474 +0xbe

--- a/kubernetes/machine_classes/alicloud-machine-class.yaml
+++ b/kubernetes/machine_classes/alicloud-machine-class.yaml
@@ -6,10 +6,10 @@ metadata:
   name: test-alicloud # Name of Alicloud machine class goes here
   namespace: default # Namespace in which the machine class is to be deployed
 spec:
-  imageName: coreos_1745_7_0_64_30G_alibase_20180705.vhd # Alicloud machine image name goes here, e.g. centos_7_04_64_20G_alibase_201701015
+  imageID: coreos_1745_7_0_64_30G_alibase_20180705.vhd # Alicloud machine image name goes here, e.g. centos_7_04_64_20G_alibase_201701015
   instanceType: ecs.n1.medium # Type of ecs machine ecs.t5-lc1m1.small ecs.t5-lc2m1.nano
   region: cn-hangzhou # Region in which machine is to be deployed
-  zone: cn-hangzhou-e # Zone of the region
+  zoneID: cn-hangzhou-e # Zone of the region
   securityGroupID: sg-1234567890 # ID of security group, it has to be within the same VPC of vSwitch
   vSwitchID: vsw-1234567890 # similar to AWS subnet ID
   systemDisk:

--- a/pkg/apis/machine/validation/alicloudmachineclass.go
+++ b/pkg/apis/machine/validation/alicloudmachineclass.go
@@ -45,10 +45,13 @@ func validateAlicloudMachineClassSpec(spec *machine.AlicloudMachineClassSpec, fl
 	allErrs := field.ErrorList{}
 
 	if "" == spec.ImageID {
-		allErrs = append(allErrs, field.Required(fldPath.Child("imageName"), "ImageName is required"))
+		allErrs = append(allErrs, field.Required(fldPath.Child("imageID"), "ImageID is required"))
 	}
 	if "" == spec.Region {
 		allErrs = append(allErrs, field.Required(fldPath.Child("region"), "Region is required"))
+	}
+	if "" == spec.ZoneID {
+		allErrs = append(allErrs, field.Required(fldPath.Child("zoneID"), "ZoneID is required"))
 	}
 	if "" == spec.InstanceType {
 		allErrs = append(allErrs, field.Required(fldPath.Child("instanceType"), "InstanceType is required"))

--- a/pkg/driver/driver_alicloud.go
+++ b/pkg/driver/driver_alicloud.go
@@ -283,7 +283,7 @@ func (c *AlicloudDriver) getEcsClient() (*ecs.Client, error) {
 	if accessKeyID != "" && accessKeySecret != "" && region != "" {
 		ecsClient, err = ecs.NewClientWithAccessKey(region, accessKeyID, accessKeySecret)
 	} else {
-		ecsClient, err = ecs.NewClient()
+		err = errors.New("alicloudAccessKeyID or alicloudAccessKeySecret can't be empty")
 	}
 	return ecsClient, err
 }

